test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,576 @@
+	admin_portal_url := "https://" + token3scale + "@" + admin_portal
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "adminportal-credentials",
+			Namespace: apicastNamespace,
+		},
+		Data: map[string][]byte{},
+	}
+	secret.Data["AdminPortalURL"] = []byte(admin_portal_url)
+	err := client.Create(ctx, secret)
+	if err != nil {
+		log.Error("Error create adminportal-credentials Secret", err)
+		return err
+	}
+	return nil
+}
+
+// Use self-managed APIcast instead of the builded one for API
+// This is attempt to automate the Promotion steps in 3scale Admin Portal
+// Require more investigation. Recommended to use manual meanwhile
+func deleteManagedApicastRoutes(ctx context.Context, client k8sclient.Client) error {
+	log.Info("Simulate Promotion in 3scale Admin Portal - Delete api-3scale-apicast- routes in " + threeScaleNamespace + " namespace")
+	routes := routev1.RouteList{}
+	err := client.List(context.TODO(), &routes, &k8sclient.ListOptions{
+		Namespace: threeScaleNamespace,
+	})
+	if err != nil {
+		log.Error("Error obtaining routes list in "+threeScaleNamespace+" namespace  ", err)
+		return err
+	}
+	for _, route := range routes.Items {
+		if strings.Contains(route.Spec.Host, "api-3scale-apicast-") {
+			err := client.Delete(ctx, &route)
+			if err != nil {
+				log.Error("error Delete Route", err)
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Install 3scale APIcast gateway operator
+func installThreeScaleApicastGatewayOperator(ctx context.Context, client k8sclient.Client) error {
+	log.Info("Install 3scale APIcast gateway Operator")
+	targetNamespaces := []string{apicastNamespace}
+	og := &v1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: apicastNamespace + "-",
+			Namespace:    apicastNamespace,
+			Generation:   1,
+			Annotations: map[string]string{
+				"olm.providedAPIs": "APIcast.v1alpha1.apps.3scale.net",
+			},
+		},
+		Spec: v1.OperatorGroupSpec{
+			TargetNamespaces: targetNamespaces,
+		},
+	}
+	err := client.Create(context.TODO(), og)
+	if err != nil && !k8serr.IsAlreadyExists(err) {
+		log.Error("Error create Operator Group in "+apicastNamespace+" namespace", err)
+		return err
+	}
+	subscription := &operatorsv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "apicast-operator",
+			Namespace: apicastNamespace,
+			Labels: map[string]string{
+				"operators.coreos.com/apicast-operator.selfmanaged-apicast": "",
+			},
+			Generation: 1,
+		},
+		Spec: &operatorsv1alpha1.SubscriptionSpec{
+			Channel:                "threescale-" + threeScaleVersion,
+			InstallPlanApproval:    operatorsv1alpha1.ApprovalAutomatic,
+			Package:                "apicast-operator",
+			CatalogSource:          "redhat-operators",
+			CatalogSourceNamespace: "openshift-marketplace",
+			StartingCSV:            "apicast-operator." + apicastOperatorVersion,
+		},
+	}
+	err = client.Create(context.TODO(), subscription)
+	if err != nil {
```

Comment:
Is it okay if the subscription already exists? I notice that you check for !IsNotFound error on the operatorGroup Create above.

Reply:
I will check it. I see that GetCondition is available to check SubscriptionCondition, but can't find any usage of this in Integrately logics

Reply:
Just to clarify, if the script is run and the Subsscription already exists in the cluster this error check with fail on `k8serr.IsAlreadyExists` Is this acceptable? I see that we handle if for the Operator Group. If it's acceptable, please feel free to resolve this conversation. 👍

Reply:
Yes, TODO, Can we handle it in https://issues.redhat.com/browse/MGDAPI-3037 ?